updater/resource.go

```diff
@@ -117,6 +117,26 @@ func (rv *ResourceVersion) isBetaVersionNumber() bool { //nolint:unused
 	}
 }
 
+// Export makes a copy of the resource with only the exposed information.
+func (res *Resource) Export() *Resource {
+	res.Lock()
```

Comment:
I'm not sure about the naming of "Export" in this case as the function declaration does not communicate what is going to happen here (it's bound to a `*Resource` and returns a `*Resource` which makes me think it's for chaining calls). I'd rather name it something like `MakePublic()` or even something like `ExposedData()`. I know we have that pattern a lot of times already and that's something that's been in my head for some time now. Feel free to close this one if you don't agree with me :)

Reply:
I understand what you mean. I left it as "Export" for now, because we use it like that in many situations.
I have improved the documentation for the function to make it clearer.